functions

```diff
@@ -14,18 +14,18 @@ parseopts() {
     longoptmatch() {
         local o longmatch=()
         for o in "${longopts[@]}"; do
-            if [[ ${o%:} = "$1" ]]; then
+            if [[ "${o%:}" == "$1" ]]; then
```

Comment:
The `==` allows pattern matching (according to `man bash`). It's not needed for string equality.

Reply:
Both `==` and `=` compare strings. The only difference is that `=` is the POSIX compatible syntax, while `==` is a bashism. Since it's a bash scripts, we should prefer bashisms.